spec/mailers/user_notifications_spec.rb

```diff
@@ -558,19 +559,49 @@
       expect(mail_html.scan(/>bobmarley/).count).to eq(1)
     end
 
-    it "the number of tags shown in subject should match max_tags_per_topic" do
-      SiteSetting.email_subject =
-        "[%{site_name}] %{optional_pm}%{optional_cat}%{optional_tags}%{topic_title}"
-      SiteSetting.max_tags_per_topic = 1
-      mail =
-        UserNotifications.user_replied(
-          user,
-          post: response,
-          notification_type: notification.notification_type,
-          notification_data_hash: notification.data_hash,
-        )
-      expect(mail.subject).to match(/Taggo/)
-      expect(mail.subject).not_to match(/Taggie/)
+    describe "the number of tags shown in subject should match max_tags_per_topic or max_tags_per_email_subject" do
```

Comment:
I think you can simplify this to `describe "number of tags shown in subject line"`

Since you outline the specific settings in describe blocks below